libs/libcds/tests/unit/libcds_flat_combining_priority_queue_stdthread.cpp

```diff
@@ -0,0 +1,111 @@
+//  Copyright (c) 2020 Weile Wei
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0.
+//  (See accompanying file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/hpx_init.hpp>
+
+#include <hpx/libcds/hpx_tls_manager.hpp>
+#include <hpx/modules/testing.hpp>
+//
+#include <cds/container/fcpriority_queue.h>
+#include <cds/init.h>    // for cds::Initialize and cds::Terminate
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <deque>
+#include <functional>
+#include <iterator>
+#include <random>
+#include <vector>
+
+static const std::size_t vec_test_size = 10000;
+
+template <class PQueue>
+void run(PQueue& pq, std::size_t i, std::vector<std::size_t>& nums)
+{
+    hpx::cds::stdthread_manager_wrapper cdswrap;
+
+    // generate a list of numbers in shuffled order
+    nums.resize(vec_test_size, 0);
+    std::iota(nums.begin(), nums.end(), i);
+    auto rng = std::default_random_engine{};
+    std::shuffle(std::begin(nums), std::end(nums), rng);
+
+    // insert random numbers into queue
+    for (auto i : nums)
+    {
+        pq.push(i);
+    }
+}
+
+// this test does not start the HPX runtime
+int main(int argc, char* argv[])
+{
+    // Initialize libcds
+    cds::Initialize();
```

Comment:
Could we have an RAII object for the global libCDS initialization as well, please?